### PR TITLE
ROX-14346: Allowing ability to add to baseline and mark as anomalous

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: ui/apps/platform/src/Containers/NetworkGraph/api/useFetchNetworkBaselines.ts [hunk 1]
```diff
@@ -23,9 +26,9 @@ const defaultResultState = {
...
  */
 function useFetchNetworkBaselines(deploymentId): FetchNetworkBaselinesResult {
-    const [result, setResult] = useState<FetchNetworkBaselinesResult>(defaultResultState);
+    const [result, setResult] = useState<Result>(defaultResultState);
 
-    useEffect(() => {
+    function fetchBaselines() {
         const networkBaselinesPromise = fetchNetworkBaselines({ deploymentId });
 
         networkBaselinesPromise
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/api/useFetchNetworkBaselines.ts:34]: Not part of this PR's changes, but could this be simplified by eliminating the "promise" var?

```suggestion
        fetchNetworkBaselines({ deploymentId })
            .then(...
```

File: ui/apps/platform/src/Containers/NetworkGraph/api/useModifyBaselineStatuses.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        status: FlowStatus,
+        onSuccessCallback: () => void
+    ) {
+        setResult({ isModifying: true, error: '' });
+        const peers = transformFlowsToPeers(flows, status);
+        const promise = markNetworkBaselineStatuses({
+            deploymentId,
+            networkBaselines: peers,
+        });
+        promise
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/api/useModifyBaselineStatuses.ts:61]: Same question as in other hook: why set the promise to a variable, just to then add the resolve handler?

File: ui/apps/platform/src/Containers/NetworkGraph/common/FlowsBulkActions.tsx [hunk 3]
```diff
@@ -7,20 +7,26 @@ type FlowsBulkActionsProps = {
...
     selectedRows,
     onClearSelectedRows,
+    markSelectedAsAnomalous,
+    addSelectedToBaseline,
 }: FlowsBulkActionsProps): ReactElement {
     // setter functions
-    const markSelectedAsAnomalous = () => {
+    const markSelectedAsAnomalousHandler = () => {
         // @TODO: Mark as anomalous
+        markSelectedAsAnomalous?.();
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/common/FlowsBulkActions.tsx:24]: Why do we need the optional chaining operator here, and immediately below?

**Review by vjwilson — APPROVED** [review]
Just a few questions to consider later

File: ui/apps/platform/src/Containers/NetworkGraph/api/useFetchNetworkBaselines.ts (hunk 1 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/api/useFetchNetworkBaselines.ts:34]: Yea we can do that

File: ui/apps/platform/src/Containers/NetworkGraph/api/useModifyBaselineStatuses.ts (hunk 2 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/api/useModifyBaselineStatuses.ts:61]: Yea that is a good point. We can remove it

File: ui/apps/platform/src/Containers/NetworkGraph/common/FlowsBulkActions.tsx (hunk 3 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/common/FlowsBulkActions.tsx:24]: I added optional chaining for now because only the baselines tab is using it. The flows tab will use in a future PR. I'm using the same FlowsTable component in both sections so I needed it to be optional for now.
